cmd/extract-block-events/main.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/fxamacker/cbor/v2"
+	"github.com/rs/zerolog"
+	"github.com/spf13/pflag"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/badger/operation"
+)
+
+func main() {
+
+	// Command line parameter initialization.
+	var (
+		flagLevel  string
+		flagData   string
+		flagBegin  uint64
+		flagFinish uint64
+		flagOutput string
+		flagSize   uint64
+		flagGroup  int
+	)
+
+	pflag.StringVarP(&flagLevel, "log-level", "l", "info", " log level for JSON logger output")
+	pflag.StringVarP(&flagData, "data-dir", "d", "data", "directory for protocol state database")
+	pflag.Uint64VarP(&flagBegin, "begin-height", "b", 0, "lowest block height to include in extraction")
+	pflag.Uint64VarP(&flagFinish, "finish-height", "f", 100_000_000, "highest block height to include in extraction")
+	pflag.StringVarP(&flagOutput, "output-dir", "o", "events", "directory for output of ledger payloads")
+	pflag.Uint64VarP(&flagSize, "size-limit", "s", 11_264_000, "limit for total size of output files")
+	pflag.IntVarP(&flagGroup, "group-size", "g", 10, "maximum number of events to extract per block")
+
+	pflag.Parse()
+
+	// Logger initialization.
+	zerolog.TimestampFunc = func() time.Time { return time.Now().UTC() }
+	log := zerolog.New(os.Stderr).With().Timestamp().Logger().Level(zerolog.DebugLevel)
+	level, err := zerolog.ParseLevel(flagLevel)
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+	log = log.Level(level)
+
+	// Initialize the protocol state database we will use.
+	opts := badger.DefaultOptions(flagData).WithLogger(nil)
+	db, err := badger.Open(opts)
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not open database")
+	}
+
+	// Initialize the codec we use for the data.
+	codec, err := cbor.CanonicalEncOptions().EncMode()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not initialize codec")
+	}
+
+	// Make a list of all available heights and shufftle them.
+	var heights []uint64
+	for height := flagBegin; height <= flagFinish; height++ {
+		heights = append(heights, height)
+	}
```

Comment:
```suggestion
	heights := make([]uint64, flagFinish+1)
	for height := range heights {
		heights = append(heights, flagBegin+height)
	}
```

Seems clearer to me this way, WDYT?

Reply:
I can't really say that your code make sense to me? O_o

You create a slice with `flagFinish+1` zeros, and then start appending to it. I guess you wanted to do something different. Personally, I find this C-style loop appropriate in this case. It explicitly says: start at flag begin, go to flag finish.

I did change the slice to be created with the initial capacity, but that's `flagFinish-flagBegin`.